kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/MockChildWorkflow.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.testing
+
+import com.squareup.workflow.StatefulWorkflow
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.stateless
+
+/**
+ * A mock implementation of [Workflow] for use in tests with `testRender` and [TestRenderResult].
+ *
+ * Note this workflow can not actually emit any output itself. Use
+ * [TestRenderResult.executeWorkflowActionFromOutput] to evaluate output handlers.
+ *
+ * @param renderer Function that is invoked in each render pass to calculate the rendering.
+ *
+ * @see StatefulWorkflow.testRender
+ * @see com.squareup.workflow.StatelessWorkflow.testRender
+ */
+class MockChildWorkflow<I, R>(private val renderer: (I) -> R) : Workflow<I, Nothing, R> {
```

Comment:
`FakeChildWorkflow`?

Reply:
It actually records the input though so you can check it later, doesn't that make it technically a "mock"?

Reply:
Yeah, Zach's right. Changed my vote. 

https://blog.pragmatists.com/test-doubles-fakes-mocks-and-stubs-1a7491dfa3da

Reply:
Oh, yes. This blog is handy 👍

Reply:
Although we lose symmetry with the worker then, which actually is a fake…

I want to add more instrumentation on them anyway, so they'll both be mocks.

Reply:
@vRallev @rjrjr  I just pushed another commit that changes the API a little. Now you call `assert*` and `execute*Output*` directly on the workers/workflows themselves, from within a block. Please take another look.